test/common/testlib.py

```diff
@@ -862,6 +862,11 @@ def cleanup_home_dirs():
             # tests expect cockpit.service to not run at start; also, avoid log leakage into the next test
             self.addCleanup(m.execute, "systemctl stop cockpit")
 
+        # The sssd daemon seems to get confused when we restore
+        # backups of /etc/group etc and stops following updates to it.
+        # Let's restart the daemon to reset that condition.
+        m.execute("systemctl try-restart sssd || true")
```

Comment:
This needs to be moved above line 843. This is a bit confusing indeed, but addCleanup() handlers are called in *reverse* order of their declaration. That usually makes sense, but here it's a bit non-obvious.

Reply:
The restart here is not a cleanup action.  Should it be?

Reply:
Interestingly [it worked](https://logs.cockpit-project.org/logs/pull-13856-20200406-121051-7beaffdb-fedora-coreos/log.html) this way around as well, but IMHO it's still cleaner to *first* restore the NSS files and *then* restart sssd.

Reply:
Oh, good point, right. It feels a little unusual, but doing it at the start of the test instead of the end should be fine as well. Sorry for the noise then!